### PR TITLE
Feat: Adds new ceph-csi integration

### PR DESCRIPTION
**Pull request by UtkarshBhatthere**
## Summary
- Add `ceph-csi` relation interface (provider side) enabling ceph-csi-operator to consume Ceph resources from MicroCeph
- Handle RBD pool creation, CephFS filesystem provisioning, and cephx auth management
- Use unit data bags for cross-model compatibility (LP#1960934)
- Add cross-model integration test suite (`tox -e ceph-csi`) using ceph-csi from `latest/edge`

## Related
- ceph-csi-operator PR: https://github.com/charmed-kubernetes/ceph-csi-operator/pull/67

## Test plan
- [x] `tox -e ceph-csi` passes locally (5/5 tests: deploy, rbd pool, auth, cephfs, cleanup)
- [ ] CI passes with ceph-csi from `latest/edge`
- [ ] Existing `tox -e unit` and `tox -e integration` tests unaffected

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/ceph_csi.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+        fs_info = json.loads(utils.run_cmd(cmd))
+
+        mdsmap = fs_info.get("mdsmap", {})
+        data_pools = mdsmap.get("data_pools")
+        metadata_pool = mdsmap.get("metadata_pool") or fs_info.get("metadata_pool")
+
+        pool_name_map = self._get_pool_name_map()
+        pools = set()
+
+        if isinstance(data_pools, list):
```
Comment (sabaini) [inline src/ceph_csi.py]: Do we expect data_pools to not be a list? Suggest to log info or warning if its not as expected

File: src/ceph_csi.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+            for pool_id in data_pools:
+                pools.add(pool_name_map.get(pool_id))
```
Comment (sabaini) [inline src/ceph_csi.py]: Are we ok with having None in `pools` if pool_id is not in the name map? This feels like it could cause issues later on. If we're sure that pool_id will always be in the map lets reference it `pool_name_map[pool_id]`

File: src/ceph_csi.py [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+            if isinstance(parsed, list):
+                items = parsed
+            elif isinstance(parsed, str):
+                items = [parsed]
+            else:
+                items = []
+        except json.JSONDecodeError:
+            items = [v.strip() for v in value.replace(";", ",").split(",")]
+
+        return {str(v).strip().lower() for v in items if str(v).strip()}
```
Comment (sabaini) [inline src/ceph_csi.py]: This workload parsing seems *very* flexible. Do we need to accept that many variants as input? Iiuc we get the workloads from the lib as json right?

**Review by sabaini — CHANGES_REQUESTED** [review]
Hey @UtkarshBhatthere thanks for the draft, left some comments

**Comment by UtkarshBhatthere** [comment]
Cancelling — this work is no longer being pursued. Closing to clean up the PR list.
